### ISSUE TITLE
Dev evironment based on conda-forge

### ISSUE THREAD

**Issue by ZzEeKkAa** [intel/intel-xpu-backend-for-triton#1552]
I'm opening this issue to track the progress on enabling project development within conda environment based on the conda-forge channel without external dependencies.

Current working solution:
1. Create environment from file:
```yaml
channels:
  - https://prefix.dev/conda-forge
  - nodefaults
dependencies:
  # complier toolchain
  - gcc_linux-64=12  # [linux]
  - gxx_linux-64=12  # [linux]
  - libgcc-ng=12
  - libstdcxx-ng=12
  - libgfortran5=12
  - libgfortran-ng=12
  - dpcpp_linux-64=2024.1.0
  # set version as close as possible to you host system
  - sysroot_linux-64=2.28 #[linux]
  # build tools
  - cmake
  - ninja
  # llvm dependencies
  - zlib
  - zstd
  - libxml2
  - python=3.10
  # enhancements
  #  enable caching between builds and for the fresh builds
  - ccache
  #  enable faster linking
  - lld
  # mkl for ipex
  - mkl-include
  # - mkl-static
  # pytorch
  - python
  - numpy<2
  - pip
  - setuptools
  - pyyaml
  - requests
  - future
  - six
  # - mkl-devel
  # - libcblas * *_mkl
  - libgomp=12
  - libabseil
  - libprotobuf
  - sleef
  - typing
  - libuv
  - pkg-config
  - typing_extensions
  # ipex
  - psutil
  - numpy
  - packaging
  # triton
  - pydot
  - pyyaml
  - matplotlib
  - numpy
  - pandas
  - textx
  - multiprocess
  # tests
  - pytest
  - pytest-xdist
  - pytest-rerunfailures
  # end triton
  # pytorch
  #- pytorch
  - pip
  - pip:
    # TODO: add mkl-static
    - mkl-static
    # triton
    - caliper-reader
    - llnl-hatchet
    # tests
    - pytest-select
    # pytorch
    #- intel-extension-for-pytorch
```

2. Build pytorch
```shell
> cd pytorch
VERBOSE=1 DEBUG=1 USE_DISTRIBUTED=1 USE_XPU=1 USE_MKLDNN=1 USE_CUDA=0 BUILD_TEST=0 USE_FBGEMM=0 USE_NNPACK=0 USE_QNNPACK=0 USE_XNNPACK=0 CC=x86_64-conda-linux-gnu-gcc CXX=x86_64-conda-linux-gnu-g++ SYCL_ROOT=$CONDA_PREFIX INTEL_MKL_DIR=$CONDA_PREFIX INTEL_COMPILER_DIR=$CONDA_PREFIX LD_LIBRARY_PATH=$CONDA_PREFIX/lib BUILD_CUSTOM_PROTOBUF=0 pip install -e . -v --no-build-isolation
```

3. Build triton
```shell
cd triton/python
VERBOSE=1 DEBUG=1 pip install -e . -v --no-build-isolation
```

4. Test triton
```shell
CPATH=$CONDA_PREFIX/include/sycl:$CPATH CC=icx CXX=icpx TRITON_TEST_SUITE=language \
pytest -vvv -n 8 --device xpu language/ --ignore=language/test_line_info.py --ignore=language/test_subprocess.py '--ignore=language/test_core.py::test_convert2d' --max-worker-restart=1000
```

## Blockers and TODOs:
- https://github.com/conda-forge/intel-compiler-repack-feedstock/issues/31
- https://github.com/conda-forge/intel-compiler-repack-feedstock/issues/32
- [ ] Add `mkl-static` to https://github.com/conda-forge/intel_repack-feedstock
- [ ] Add `caliper-reader`, `llnl-hatchet`, `pytest-select` to `conda-forge`
- [ ] Avoid loading nvidia compiler from `setup.py`
- [ ] Pick up level-zero package from conda environment (currently it pulls and builds)
- [ ] Pick up intel's compiler and libraries from conda environment. Current workaround: `export CPATH=$CONDA_PREFIX/include/sycl:$CPATH CC=icx CXX=icpx`

## Nice to haves
- [ ] Different versions [IGC](https://github.com/conda-forge/intel-graphics-compiler-feedstock) to test updates instantly

## Workarounds for the upstream

These two PRs need to be apply to achieve same compatibility and performance with IPEX:

- https://github.com/pytorch/pytorch/pull/126456
- https://github.com/pytorch/pytorch/pull/129675

## Other issues

- If you have ssl version mismatch during build, install `openssh` conda package.

**Comment by pbchekin** [comment]
Is it for build or runtime environment?

**Comment by pbchekin** [comment]
CC @leshikus

**Comment by ZzEeKkAa** [comment]
> Is it for build or runtime environment?

For development environment, in other words for both.

**Comment by vlad-penkin** [comment]
@leshikus, @pbchekin, @ZzEeKkAa PTDB requires special ABI neutral mkl build which is not distributed via intel (`-c https://software.repos.intel.com/python/conda/` ) or conda-forge channels. Let's revisit this ticket after 2025.0 is released.

**Comment by ZzEeKkAa** [comment]
I agree that in current state it is blocked by 2025.0 release.

**Comment by ZzEeKkAa** [comment]
2025 is coming soon, but there is actually a workaround to compile pytorch with `USE_STATIC_MKL=1`. However it must be pulled from PTDB distribution.

**Comment by ZzEeKkAa** [comment]
I've updated blockers with missing pti-gpu distribution.
